docs/DeveloperGuide.adoc

```diff
@@ -218,29 +328,72 @@ image::CommitActivityDiagram.png[]
 
 ===== Aspect: How undo & redo executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Individual command knows how to undo/redo by itself.
+** Pros: Will use less memory (e.g. for `delete`, just save the book being deleted).
+** Cons: We must ensure that the implementation of each individual command are correct.
+* **Alternative 2:** Saves the entire catalog.
 ** Pros: Easy to implement.
 ** Cons: May have performance issues in terms of memory usage.
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
 
 ===== Aspect: Data structure to support the undo/redo commands
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+* **Alternative 1 (current choice):** Use two stacks to store a list of undoable and redoable commands.
+** Pros: Easy for future developers to understand as there are two separate stacks to keep track of the command to undo and redo.
+** Cons: Additional time required to add and pop from the stack.
+* **Alternative 2:** Use a list to store the commands for undo and redo.
+** Pros: We do not need to maintain a separate stack.
+** Cons: Harder for new developers to understand.
+// end::undo/redo[]
+
+// tag::printLoanSlip[]
+=== Generate Loan Slip feature
+==== Proposed Implementation
+
+The printing of loan slip feature is facilitated by `LoanSlipUtil`.
+Essentially, it implements the following operations:
+
+* `LoanSlipUtil#mountLoan()` -- Mounts a loan slip in preparation for generation of loan slip in pdf form.
+* `LoanSlipUtil#unmountLoan()` -- Unmounts a loan slip ,usually after generating a pdf version of it.
+* `LoanSlipUtil#createLoanSlipInDirectory()` -- Creates a pdf version of the mounted loan slip in the loan_slips folder.
+
+Given below is the sequence diagram of the generation of loan slip during the loan of a book.
+
+.Sequence Diagram for the generation of a loan slip
+image::LoanSlipGeneration.png[]
```

Comment:
The return arrow on a sequence diagram should be dotted. There's a way to place dotted lines using PlantUml!

Reply:
Good catch! Will make ammendments to it